### DISCUSSION TITLE
[Feature Request] Add the ability to restart an interpreter without having to re-add all listeners

### DISCUSSION THREAD

**Discussion by AlCalzone** [statelyai/xstate#1484] [General]
I'm looking for a simpler way to restart an interpreter. `stop()` removes all listeners, so if I want to keep using the interpreter, I have to re-add them before starting. It would be nice to have a `restart()` method to achieve that.

**Comment by davidkpiano** [comment]
Can you give me a real-world use-case? The way you should do this is have the root machine in some sort of "paused" state.

**Reply by AlCalzone** [reply]
The real-world use case comes from my library node-zwave-js. I'm representing both the status (alive, dead, asleep, awake, unknown) and the "readyness" (not ready, maybe ready, definitely ready) of devices with an interpreted state machine.

The user of this library can re-interview a device. This requires the library to forget most of the information about that node, including the readyness and status.

I'm now doing it like this:
Create the interpreters and listeners: https://github.com/AlCalzone/node-zwave-js/blob/0ad943969ccc18ef0ad1732379c5d3bc02ef7022/packages/zwave-js/src/lib/node/Node.ts#L191-L225
Reset the interpreters when necessary: https://github.com/AlCalzone/node-zwave-js/blob/0ad943969ccc18ef0ad1732379c5d3bc02ef7022/packages/zwave-js/src/lib/node/Node.ts#L811-L812
but that is not very elegant and requires two non-null assertions because I no longer create the interpreters in the constructor.

**Reply by Andarist** [reply]
You could just create a wrapper over `interpret` that would re-add your listeners before each `start`. The facade patterns is very powerful for custom needs like this - we can't just cover all potential use cases on our own, it's not practical.
